### PR TITLE
Navigation buttons hideability during installation

### PR DESCRIPTION
**Pull request by deprov447**
### Addressing #1388

### CODE REVIEW

**Comment by deprov447** [comment]
@adriaandegroot Please review the PR

File: src/libcalamares/Settings.h [hunk 1]
```diff
@@ -176,6 +178,7 @@ class DLLEXPORT Settings : public QObject
     bool m_promptInstall;
     bool m_disableCancel;
     bool m_disableCancelDuringExec;
+    bool m_hideBackAndNextDuringExec;
```
Comment (adriaandegroot) [inline src/libcalamares/Settings.h]: ` = false;` or it needs to be initialized in the constructors in `Settings.cpp`. Come to think of it, each of these bools could be `=` assigned here in the class declaration, and the initializer lists in the constructor simplified.

File: src/libcalamaresui/ViewManager.cpp [hunk 2]
```diff
@@ -369,6 +369,7 @@ ViewManager::next()
             UPDATE_BUTTON_PROPERTY( backEnabled, false )
         }
         updateCancelEnabled( !settings->disableCancel() && !( executing && settings->disableCancelDuringExec() ) );
+        updateBackAndNextVisibility(!executing || !settings->hideBackAndNextDuringExec());
```
Comment (adriaandegroot) [inline src/libcalamaresui/ViewManager.cpp]: Apply the coding-style tool (it will put in some more spaces).

**Comment by adriaandegroot** [comment]
Overall this looks really good. The review (above) just has a nitpick (that can be mechanically resolved by running the tool) and a could-be-better that is outside the scope of your PR. 

One thing I'm missing, though: connecting this to the buttons in the non-QML navigation panel. `CalamaresWindow::getWidgetNavigation( QWidget* parent )` is the function that creates the nav-panel, and you'll want to hook up to the ViewManager signals there (to the back and next buttons).

(Medium-term I'd like to reconsider whether this setting, and similar settings like show-quit, belong in `settings.conf` or in the Branding; I have vaguely in mind that Settings is overall functionality and Branding steers the UI, but that's not done consistently yet and it'd be a Calamares 3.3 thing anyway)

File: src/libcalamares/Settings.h (hunk 1 above)
Comment (deprov447) [reply src/libcalamares/Settings.h]: Okay. I've looked into the default `settings.conf` file and tried to extract values from there. Kindly just **verify** them once.

File: src/libcalamaresui/ViewManager.cpp (hunk 2 above)
Comment (deprov447) [reply src/libcalamaresui/ViewManager.cpp]: Done

**Comment by deprov447** [comment]
> Overall this looks really good. The review (above) just has a nitpick (that can be mechanically resolved by running the tool) and a could-be-better that is outside the scope of your PR.
> 
> One thing I'm missing, though: connecting this to the buttons in the non-QML navigation panel. `CalamaresWindow::getWidgetNavigation( QWidget* parent )` is the function that creates the nav-panel, and you'll want to hook up to the ViewManager signals there (to the back and next buttons).
> 
> (Medium-term I'd like to reconsider whether this setting, and similar settings like show-quit, belong in `settings.conf` or in the Branding; I have vaguely in mind that Settings is overall functionality and Branding steers the UI, but that's not done consistently yet and it'd be a Calamares 3.3 thing anyway)

@adriaandegroot I tried hooking up the signal from `ViewManager.cpp` to the Navigation widget but was unable to do so. Can you provide some hint regarding that?

And yes some of the variables in `settings.conf` are more of a branding opinion and should live there, in `branding.desc`. If you want I can pull the input values from there.

**Comment by deprov447** [comment]
@adriaandegroot Please review.
Commit [a953901](https://github.com/calamares/calamares/pull/1590/commits/a9539018e968a535557c604f4bdb0165d701287c) fixes a small boolean error.
Commit [0ff3278](https://github.com/calamares/calamares/pull/1590/commits/0ff32784d11f1100305180d36eeabb955a0cb5bb) hooks backAndNextVisible signal from ViewManager.cpp to CalamaresWindow.

**Pull request merged**